File: LOO 2°Parcial/Tarea 4/TrabajadorPorHoras.py
# -*- coding: utf-8 -*-
"""
Created on Mon Apr 12 10:53:33 2021

@author: MedNo
"""

import Empleado as Emp

class TrabajadorPorHoras(Emp.Empleado):
    def __init__(self, nombre="", cantidad=0, pago=0):
        Emp.Empleado.__init__(self,nombre)
        if cantidad>0:
            self.__HorasTrabajadas = cantidad
        else:
            self.__HorasTrabajadas = 0
        if pago > 0:
            self.__pagoXhora = pago
        else:
            self.__pagoXhora = 0
            
    @property 
    def HorasTrabajadas(self):
        return self.__HorasTrabajadas
    @HorasTrabajadas.setter
    def HorasTrabajadas(self, cantidad):
        if cantidad>0:
            self.__HorasTrabajadas = cantidad
        else:
            self.__HorasTrabajadas = 0
    @property
    def pagoXhora(self):
        return self.__pagoXhora
    @pagoXhora.setter
    def pagoXhora(self, pago):
        if pago > 0:
            self.__pagoXhora = pago
        else:
            self.__pagoXhora = 0
            
    def pagoDiario(self):
        return self.__HorasTrabajadas * self.__pagoXhora
    
    def imprimir(self):
        print("Empleado tipo Trabajador por horas");
        print("NOMBRE: " + self.nombre + "\nPAGO POR HORA: " + str(self.__pagoXhora) + "\nCANTIDAD HORAS TRABAJADAS: " + str(self.__HorasTrabajadas) + "\nPAGO DIARIO: " + str(self.pagoDiario()))
   

File: LOO 2°Parcial/EmpleadosCSharp/EmpleadosCSharp/TrabajadorPorHoras.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmpleadosCSharp
{
    class TrabajadorPorHoras : Empleado
    {
        protected int HorasTrabajadas;
        protected double pagoXhora;
        public TrabajadorPorHoras() : base()
        {
            this.HorasTrabajadas = 0;
            this.pagoXhora = 0;
        }
        public TrabajadorPorHoras(String nombre, int cantidad, double pago) : base(nombre)
        {
            this.HorasTrabajadas = cantidad > 0 ? cantidad : 0;
            this.pagoXhora = pago > 0 ? pago : 0;
        }
        public int HrsTrabajadas
        {
            set { this.HorasTrabajadas = value > 0 ? value : 0; }
            get { return this.HorasTrabajadas; }
        }
        public double PagoxHora
        {
            set { this.pagoXhora = value > 0 ? value : 0; }
            get { return this.pagoXhora; }
        }
        public override double pagoDiario()
        {
            return this.HorasTrabajadas * this.pagoXhora;
        }
        public override void imprimir()
        {
            Console.WriteLine("Empleado tipo Trabajador por horas");
            Console.WriteLine("NOMBRE: " + this.nombre + "\nPAGO POR HORA: " + this.pagoXhora + "\nCANTIDAD HORAS TRABAJADAS: " + this.HorasTrabajadas + "\nPAGO DIARIO: " + this.pagoDiario());
        }
    }
}
